[docs] Split Expo Skills table into framework and EAS sections (#47599)
expo/expo: docs/pages/skills.mdx

## Available Expo Skills

> The following skills are available in the `expo` plugin. Skill names describe what they cover: `expo-*` skills teach agents the open source framework, and `eas-*` skills cover EAS services.

Author: ```suggestion
The following skills are available in the `expo` plugin. Skills name prefixed with`expo-*` skills are used for the Expo open source framework and `eas-*` skills for the EAS services.
```
